src/app/pages/apps/store/docker.service.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  computed, Injectable,
+} from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { Observable, switchMap, tap } from 'rxjs';
+import { DockerConfig, DockerStatusResponse } from 'app/enums/docker-config.interface';
+import { DockerStatus } from 'app/enums/docker-status.enum';
+import { ApiEvent } from 'app/interfaces/api-message.interface';
+import { WebSocketService } from 'app/services/ws.service';
+
+export interface DockerConfigState {
+  isLoading: boolean;
+  pool: string;
+  statusData: {
+    status: string;
+    description: string;
+  };
+}
+
+const initialState: DockerConfigState = {
+  isLoading: false,
+  pool: null,
+  statusData: {
+    status: null,
+    description: null,
+  },
+};
+
+@Injectable()
+export class DockerStore extends ComponentStore<DockerConfigState> {
+  readonly isLoading = computed(() => {
+    return this.state().isLoading;
+  });
+
+  readonly selectedPool = computed(() => {
+    return this.state().pool;
+  });
+
+  readonly dockerStarted = computed(() => {
+    return [
+      DockerStatus.Initializing,
+      DockerStatus.Pending,
+      DockerStatus.Running,
+    ].includes(this.state().statusData.status as DockerStatus);
+  });
+
+  readonly status = computed(() => {
+    return this.state().statusData.status;
+  });
+
+  readonly statusDescription = computed(() => {
+    return this.state().statusData.description;
+  });
+
+  constructor(
+    private ws: WebSocketService,
+  ) {
+    super(initialState);
+    this.initialize();
```

Comment:
Maybe call this explicitly in `AppsScopeWrapperComponent`, so that there is no unexpected behaviour for other developers.